test/extensions/transport_sockets/http_11_proxy/connect_test.cc

```diff
@@ -0,0 +1,353 @@
+
+class Http11ConnectTest : public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  Http11ConnectTest() {}
+  void initialize(bool no_proxy_protocol = false) {
+    std::string address_string =
+        absl::StrCat(Network::Test::getLoopbackAddressUrlString(GetParam()), ":1234");
+    auto address = Network::Utility::parseInternetAddressAndPort(address_string);
+    auto info =
+        std::make_unique<Network::TransportSocketOptions::ProxyInfo>("www.foo.com", address);
+    if (no_proxy_protocol) {
+      info.reset();
+    }
+
+    options_ = std::make_shared<const Network::TransportSocketOptionsImpl>(
+        "", std::vector<std::string>{}, std::vector<std::string>{}, std::vector<std::string>{},
+        absl::nullopt, nullptr, std::move(info));
+
+    setAddress();
+    auto inner_socket = std::make_unique<NiceMock<Network::MockTransportSocket>>();
+    inner_socket_ = inner_socket.get();
+    EXPECT_CALL(*inner_socket_, ssl()).Times(AnyNumber()).WillRepeatedly(Return(ssl_));
+    EXPECT_CALL(Const(*inner_socket_), ssl()).Times(AnyNumber()).WillRepeatedly(Return(ssl_));
+
+    ON_CALL(transport_callbacks_, ioHandle()).WillByDefault(ReturnRef(io_handle_));
+    connect_socket_ =
+        std::make_unique<UpstreamHttp11ConnectSocket>(std::move(inner_socket), options_);
+    connect_socket_->setTransportSocketCallbacks(transport_callbacks_);
+    connect_socket_->onConnected();
+  }
+
+  void setAddress() {
+    std::string address_string =
+        absl::StrCat(Network::Test::getLoopbackAddressUrlString(GetParam()), ":1234");
+    auto address = Network::Utility::parseInternetAddressAndPort(address_string);
+    transport_callbacks_.connection_.stream_info_.filterState()->setData(
+        "envoy.network.transport_socket.http_11_proxy.address",
+        std::make_unique<Network::FilterStateProxyInfo>("www.foo.com", address),
+        StreamInfo::FilterState::StateType::ReadOnly,
+        StreamInfo::FilterState::LifeSpan::FilterChain);
+  }
+
+  Network::TransportSocketOptionsConstSharedPtr options_;
+  NiceMock<Network::MockTransportSocket>* inner_socket_;
+  NiceMock<Network::MockIoHandle> io_handle_;
+  std::unique_ptr<UpstreamHttp11ConnectSocket> connect_socket_;
+  NiceMock<Network::MockTransportSocketCallbacks> transport_callbacks_;
+  Buffer::OwnedImpl connect_data_{"CONNECT www.foo.com:443 HTTP/1.1\r\n\r\n"};
+  Ssl::ConnectionInfoConstSharedPtr ssl_{
+      std::make_shared<NiceMock<Envoy::Ssl::MockConnectionInfo>>()};
+};
+
+// Test injects CONNECT  only once
+TEST_P(Http11ConnectTest, InjectesHeaderOnlyOnce) {
+  initialize();
+
+  EXPECT_CALL(io_handle_, write(BufferStringEqual(connect_data_.toString())))
+      .WillOnce(Invoke([&](Buffer::Instance& buffer) -> Api::IoCallUint64Result {
+        auto length = buffer.length();
+        buffer.drain(length);
+        return Api::IoCallUint64Result(length, Api::IoErrorPtr(nullptr, [](Api::IoError*) {}));
+      }));
+  auto msg = Buffer::OwnedImpl("initial data");
```

Comment:
How about `Buffer::OwnedImpl msg("initial data")` which is shorter and does not use auto. (here and elsewhere)

Reply:
done